vpr/src/place/initial_placement.cpp

```diff
@@ -597,7 +597,23 @@ static bool try_centroid_placement(const t_pl_macro& pl_macro,
         const auto& compressed_block_grid = g_vpr_ctx.placement().compressed_block_grids[block_type->index];
         const auto& type = device_ctx.grid.get_physical_type({centroid_loc.x, centroid_loc.y, centroid_loc.layer});
         const auto& compatible_sub_tiles = compressed_block_grid.compatible_sub_tile_num(type->index);
-        centroid_loc.sub_tile = compatible_sub_tiles[rng.irand((int)compatible_sub_tiles.size() - 1)];
+        
```

Comment:
It seems like this code should be hoisted earlier; maybe as a helper function called in both find_centroid_loc and find_centroid_neighbor.  It seems strange that we find a compatible subtile only if we didn't have to do a search for a neighbour; I think we should find a compatible and empty subtile whenever we return a centroid location, or else we should return an invalid one.  Or, if we can't do that because some users of the find_centroid_loc don't care if the location is occupied or not we should add a call to a helper function to figure out what the unoccupied sub_tile would be right after calling the find_centroid* routines.